### PR TITLE
✨ Add raw results to cron scans

### PR DESCRIPTION
**Pull request by laurentsimon**
Adds the raw results to cron jobs
no breaking changes

Note: I have not created the buckets and tables yet. I'll do more testing but would like to get reviews in the meantime.

```release-notes
Add raw results to the weekly cron
```

### CODE REVIEW

File: cron/k8s/raw-transfer-v0.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+          - name: bq-raw-transfer-v0
+            image: gcr.io/openssf/scorecard-bq-transfer:latest
+            imagePullPolicy: Always
+            resources:
+              limits:
+                memory: 1Gi
+              requests:
+                memory: 1Gi
+            env:
+              - name: RAW_SCORECARD_BIGQUERY_TABLEV0
```
Comment (laurentsimon) [inline cron/k8s/raw-transfer-v0.yaml:37]: I will comment out this file when merging to only perform the release-test

File: cron/config/config.go [hunk 2]
```diff
@@ -47,9 +47,12 @@ const (
 	metricExporter         string = "SCORECARD_METRIC_EXPORTER"
 	ciiDataBucketURL       string = "SCORECARD_CII_DATA_BUCKET_URL"
 	blacklistedChecks      string = "SCORECARD_BLACKLISTED_CHECKS"
-
+	// v2 results.
 	bigqueryTableV2       string = "SCORECARD_BIGQUERY_TABLEV2"
 	resultDataBucketURLV2 string = "SCORECARD_DATA_BUCKET_URLV2"
+	// v0 raw results.
+	rawBigqueryTableV0       string = "RAW_SCORECARD_BIGQUERY_TABLEV0"
+	rawResultDataBucketURLV0 string = "RAW_SCORECARD_DATA_BUCKET_URLV0"
```
Comment (azeemshaikh38) [inline cron/config/config.go:55]: Remove the `v0` here and anywhere else. It unnecessarily adds a code cleanup task later.

File: cron/format/json_raw_results.go [hunk 3]
```diff
@@ -0,0 +1,308 @@
+// Copyright 2022 Security Scorecard Authors
```
Comment (azeemshaikh38) [inline cron/format/json_raw_results.go:1]: Any reason there is separate logic for this in `cron`? This should ideally be the same JSON as exposed to users in `checker` package.

File: cron/k8s/raw-transfer-v0.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
+# Copyright 2021 Security Scorecard Authors
```
Comment (azeemshaikh38) [inline cron/k8s/raw-transfer-v0.yaml:1]: Use the file naming convention - `transfer-raw.yaml`. Remove `v0`.

File: cron/k8s/transfer.raw-release-v0.yaml [hunk 5]
```diff
@@ -0,0 +1,43 @@
+# Copyright 2021 Security Scorecard Authors
```
Comment (azeemshaikh38) [inline cron/k8s/transfer.raw-release-v0.yaml:1]: Use filenaming convention - `transfer.release-raw.yaml`

File: cron/k8s/controller.release.yaml [hunk 6]
```diff
@@ -69,6 +69,16 @@ spec:
...
+            # Raw results v0.
+            - name: RAW_SCORECARD_DATA_BUCKET_URLV0
+              value: "gs://ossf-scorecard-rawdata-releasetest-v0"
+            - name: SCORECARD_SHARD_SIZE
+              value: "5"
+            resources:
+              limits:
+                memory: 1Gi
+              requests:
+                memory: 1Gi
```
[the remark this answers is not in the document]
Comment (laurentsimon) [reply cron/k8s/controller.release.yaml:81]: oops, thanks

File: cron/config/config.go (hunk 2 above)
Comment (laurentsimon) [reply cron/config/config.go:55]: done

File: cron/k8s/raw-transfer-v0.yaml (hunk 4 above)
Comment (laurentsimon) [reply cron/k8s/raw-transfer-v0.yaml:1]: done, thanks

File: cron/k8s/transfer.raw-release-v0.yaml (hunk 5 above)
Comment (laurentsimon) [reply cron/k8s/transfer.raw-release-v0.yaml:1]: done, thanks

File: cron/format/json_raw_results.go (hunk 3 above)
Comment (laurentsimon) [reply cron/format/json_raw_results.go:1]: like the existing json-based results, this gives us a chance to update the raw result format for the scorecard, without breaking the cron format. For now this is simply going to be the same code, though

**Comment by laurentsimon** [comment]
I can merge this today, so long as it won't impact the weekly scan. Can you confirm the weekly scan only uses hash commit that successfully passed the releasetest?

File: cron/format/json_raw_results.go (hunk 3 above)
Comment (azeemshaikh38) [reply cron/format/json_raw_results.go:1]: > update the raw result format for scorecard, without breaking the cron format. 

Yeah, but the raw JSON schema will be publicly available and the cron job should honor that.

Ok for now, but ideally this shouldn't diverge.

**Review by azeemshaikh38 — APPROVED** [review]

**Comment by azeemshaikh38** [comment]
> I can merge this today, so long as it won't impact the weekly scan. Can you confirm the weekly scan only uses hash commit that successfully passed the releasetest?

Feel free to merge this in. You'll need to to `kubectl apply -f` locally (we don't have this automated yet). It will get tested in releasetest first before running in the prod cron job.

**Pull request merged**